contrib/cmd/wait-for-cluster-ready/waitforcluster.go

```diff
@@ -23,130 +23,500 @@ import (
 		}
 		if err != nil {
-			return false, err
+			return false, fmt.Errorf("Unexpected error retrieving cluster %s/%s: %v", o.Namespace, name, err)
 		}
+		log.Debugf("Cluster resource was found")
 		return true, nil
 	})
+	if err == nil {
+		log.Infof("cluster %s/%s has been created.", o.Namespace, name)
+	} else if err == wait.ErrWaitTimeout {
+		err = fmt.Errorf("timed out waiting %v for cluster %s/%s to be created.", o.ClusterResourceTimeout, o.Namespace, name)
+	}
+	return err
+}
+
+func (o *WaitForClusterOptions) waitForInfrastructure(client capiclient.Interface, name string) error {
+	log.Infof("Waiting for infrastructure provisioning for cluster %s/%s", o.Namespace, name)
+	startedProvision := false
+	return o.waitForClusterStatus(client, name, o.InfraProvisioningTimeout, func(status *coapiv1.ClusterProviderStatus) bool {
+		if status.Provisioned {
+			log.Infof("infrastructure provisioned for cluster %s/%s", o.Namespace, name)
+			return true
+		}
+		provisioning := controller.FindClusterCondition(status.Conditions, coapiv1.ClusterInfraProvisioning)
+		if provisioning != nil && !startedProvision && provisioning.Status == corev1.ConditionTrue {
+			startedProvision = true
+			log.Infof("infrastructure started provisioning on cluster %s/%s at %v", o.Namespace, name, provisioning.LastTransitionTime)
+		}
+		return false
+	})
+}
+
+func (o *WaitForClusterOptions) waitForMasterInstall(client capiclient.Interface, name string) error {
+	log.Infof("Waiting for master install on cluster %s/%s", o.Namespace, name)
+	startedInstall := false
+	return o.waitForClusterStatus(client, name, o.MasterInstallTimeout, func(status *coapiv1.ClusterProviderStatus) bool {
+		if status.ControlPlaneInstalled {
+			log.Infof("control plane installed on cluster %s/%s", o.Namespace, name)
+			return true
+		}
+		installing := controller.FindClusterCondition(status.Conditions, coapiv1.ControlPlaneInstalling)
+		if installing != nil && !startedInstall && installing.Status == corev1.ConditionTrue {
+			startedInstall = true
+			log.Infof("control plane started installing on cluster %s/%s at %v", o.Namespace, name, installing.LastTransitionTime)
+		}
+		return false
+	})
+}
+
+func (o *WaitForClusterOptions) waitForClusterAPI(client capiclient.Interface, name string) error {
+	log.Infof("Waiting for cluster API install on cluster %s/%s", o.Namespace, name)
+	startedInstall := false
+	return o.waitForClusterStatus(client, name, o.ClusterAPIInstallTimeout, func(status *coapiv1.ClusterProviderStatus) bool {
+		if status.ClusterAPIInstalled {
+			log.Infof("cluster API installed on cluster %s/%s", o.Namespace, name)
+			return true
+		}
+		installing := controller.FindClusterCondition(status.Conditions, coapiv1.ClusterAPIInstalling)
+		if installing != nil && !startedInstall && installing.Status == corev1.ConditionTrue {
+			startedInstall = true
+			log.Infof("cluster API started installing on cluster %s/%s at %v", o.Namespace, name, installing.LastTransitionTime)
```

Comment:
Would suggest just a quick scan of the output for consistency, specifically there's a mix of starting case for each line and trailing punctuation. No pref for either just would look a little cleaner if they were all the same.